### PR TITLE
fix: 🐛 Use latest UIs if version specific UIs are missing

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

We generally don't publish UIs compatible with each version, but the workflow tries to download the version specific UIs. This change adds a fallback flow where latest UIs are used to spin up the instance if version specific UIs are not found.

### Breaking Changes

NA

### JIRA Link

NA

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/common/util.ts [hunk 1]
```diff
@@ -91,7 +91,12 @@ export function printInfo(cmd: Command): void {
 export async function downloadFile(url: string, dest: string): Promise<void> {
   const destDir = path.dirname(dest);
   mkdirpSync(destDir); // ensure dir exists
-  await promisify(pipeline)((await fetch(url)).body, createWriteStream(dest));
+  const response = await fetch(url);
+  if (response.status === 200) {
```
Comment (polymath-eric) [inline src/common/util.ts:95]:
```suggestion
  if (response.status === 200 && response.body) {
```
I think its an error if there is nothing being written?

File: src/common/util.ts [hunk 2]
```diff
@@ -91,7 +91,12 @@ export function printInfo(cmd: Command): void {
...
+    await promisify(pipeline)(response?.body, createWriteStream(dest));
+  } else {
+    throw new Error('Could not download file');
```
Comment (polymath-eric) [inline src/common/util.ts:98]:
```suggestion
    throw new Error('Could not download file');
```
```suggestion
    throw new Error(`Could not download file: ${url}`);
```

File: src/common/util.ts [hunk 3]
```diff
@@ -91,7 +91,12 @@ export function printInfo(cmd: Command): void {
 export async function downloadFile(url: string, dest: string): Promise<void> {
   const destDir = path.dirname(dest);
   mkdirpSync(destDir); // ensure dir exists
-  await promisify(pipeline)((await fetch(url)).body, createWriteStream(dest));
+  const response = await fetch(url);
+  if (response.status === 200) {
+    await promisify(pipeline)(response?.body, createWriteStream(dest));
```
Comment (polymath-eric) [inline src/common/util.ts:96]:
```suggestion
    await promisify(pipeline)(response.body, createWriteStream(dest));
```

**Review by polymath-eric — APPROVED** [review]

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**